src/Forms/GridField/GridFieldAddExistingAutocompleter.php

```diff
@@ -283,12 +286,15 @@ public function doSearch($gridField, $request)
         $json = [];
         Config::nest();
         SSViewer::config()->set('source_file_comments', false);
-        $viewer = SSViewer::fromString($this->resultsFormat);
+
+        $engine = Injector::inst()->create(TemplateEngine::class);
```

Comment:
```suggestion
        $engine = Injector::inst()->get(TemplateEngine::class);
```
Should we be using a singleton instead? Same question applies for any other other times we're doing this in this PR

Reply:
Discussion about this can be held in https://github.com/silverstripe/silverstripe-framework/pull/11405#discussion_r1800332061